plugin/al/lib/AL_USDMaya/AL/usdmaya/nodes/ProxyShape.cpp

```diff
@@ -1637,7 +1637,9 @@ bool ProxyShape::lockTransformAttribute(const SdfPath& path, const bool lock)
   r.setLocked(lock);
   s.setLocked(lock);
 
-  if (lock && MFnDependencyNode(lockObject).typeId() == AL_USDMAYA_TRANSFORM)
+  MFnDependencyNode fn(lockObject);
+  Scope* transformNode = static_cast<Scope*>(fn.userNode());
```

Comment:
Should this one and the one on line 2144 be dynamic_cast?

Reply:
I think you're right -It would be OK to use static_cast if the data had been pre-filtered to only allow Scopes or classes derived from it,  or we were doing some kind of checking earlier on in the method, but as "lockObject" can be any maya object  we should either be doing those checks or using a dynamic_cast to check that it's a valid Scope at run -time... (add some minor performance cost). Will update. Thanks!